homeworks/roman.buzukov_BuzRom/3-async/js/script.js

```diff
@@ -0,0 +1,86 @@
+const containerEl = document.querySelector('[data-container]');
+const listEl = document.querySelector('[data-list]');
+const templateEl = document.querySelector('[data-post-template]');
+const sortEl = document.querySelector('[data-sort]');
+const filterEl = document.querySelector('[data-filter]');
+const removeBtn = document.getElementsByClassName('post__delete');
+const dataPosts = [];
+let posts = dataPosts;
+
+const deletePost = (e) => {
+    if (e.target) {
+        const post = e.target.closest('[data-post]');
+        post.setAttribute('style', 'visibility: hidden');
+        containerEl.classList.remove('loaded');
+        setTimeout(() => {
+            fetch(`https://jsonplaceholder.typicode.com/posts/${post.id}`, {
+                method: 'DELETE',
+            })
+                .then((response) => response.json())
+                .then(() => {
+                    alert('Post deleted');
+                    posts = posts.filter((item) => +post.id !== item.id);
+                    post.remove();
+                })
+                .catch(() => {
+                    alert('Something went wrong');
+                    post.setAttribute('style', 'visibility: visible');
+                });
+            containerEl.classList.add('loaded');
+        }, 1000);
+    }
+};
+
+const addDeleteListener = (deleteEl) => {
+    Array.from(deleteEl).forEach((item) => {
+        item.addEventListener('click', deletePost);
+    });
+};
+
+const renderPosts = (postsArray) => {
+    listEl.innerHTML = '';
+    postsArray.map((element) => {
+        const item = templateEl.content.cloneNode(true);
+        item.querySelector('[data-post]').setAttribute('id', `${element.id}`);
+        item.querySelector('[data-title]').textContent = element.title;
+        item.querySelector('[data-body]').textContent = element.body;
+        listEl.appendChild(item);
+        return addDeleteListener(removeBtn);
```

Comment:
Almost finished! But you're passing the global CSS buttons list.
I suggest you get `itemRemoveEl` from `item.querySelector`, and pass it here as an argument.

Also, you need to update `addDeleteListener`. It should receive **one** element and add an event to it.

Reply:
Now I get it!
It's have sense 👍!

Reply:
Thank you!